openedx/core/djangoapps/discussions/tasks.py

```diff
@@ -85,11 +85,10 @@ def iter_discussable_units():
 
     with store.branch_setting(ModuleStoreEnum.Branch.published_only, course_key):
         course = store.get_course(course_key)
-        provider = course.discussions_settings.get('provider', provider_type)
-        enable_in_context = course.discussions_settings.get('enable_in_context', True)
-        provider_config = course.discussions_settings.get(provider, {})
-        unit_level_visibility = course.discussions_settings.get('unit_level_visibility', True)
-        enable_graded_units = course.discussions_settings.get('enable_graded_units', False)
+        enable_in_context = discussions_config.enable_in_context
+        provider_config = discussions_config.plugin_configuration
+        unit_level_visibility = discussions_config.unit_level_visibility
+        enable_graded_units = discussions_config.enable_graded_units
```

Comment:
The rest of the code uses these settings from the database, so we do as well here.